cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -32,38 +33,46 @@ import (
 type Reporter struct {
 	collector api_v2.CollectorServiceClient
 	logger    *zap.Logger
+	sanitizer zipkin2.Sanitizer
 }
 
 // NewReporter creates gRPC reporter.
 func NewReporter(conn *grpc.ClientConn, logger *zap.Logger) *Reporter {
+	zSanitizer := zipkin2.NewChainedSanitizer(
+		zipkin2.NewSpanDurationSanitizer(),
+		zipkin2.NewSpanStartTimeSanitizer(),
+		zipkin2.NewParentIDSanitizer(),
+		zipkin2.NewErrorTagSanitizer(),
+	)
 	return &Reporter{
 		collector: api_v2.NewCollectorServiceClient(conn),
 		logger:    logger,
+		sanitizer: zSanitizer,
 	}
 }
 
 // EmitBatch implements EmitBatch() of Reporter
 func (r *Reporter) EmitBatch(b *thrift.Batch) error {
-	// TODO pass process to r.send() - do not convert it for every span
-	spans := jConverter.ToDomain(b.Spans, b.Process)
-	return r.send(spans)
+	return r.send(jConverter.ToDomain(b.Spans, nil), jConverter.ToDomainProcess(b.Process))
 }
 
 // EmitZipkinBatch implements EmitZipkinBatch() of Reporter
 func (r *Reporter) EmitZipkinBatch(zSpans []*zipkincore.Span) error {
+	for _, zSpan := range zSpans {
+		zSpan = r.sanitizer.Sanitize(zSpan)
+	}
 	trace, err := zipkin.ToDomain(zSpans)
 	if err != nil {
 		return err
 	}
-	return r.send(trace.Spans)
+	return r.send(trace.Spans, nil)
 }
 
-func (r *Reporter) send(spans []*model.Span) error {
-	var process model.Process
-	if len(spans) > 0 {
-		process = *spans[0].Process
+func (r *Reporter) send(spans []*model.Span, process *model.Process) error {
+	batch := model.Batch{Spans: spans}
+	if process != nil {
+		batch.Process = *process
```

Comment:
oh-oh, I think I made a mistake in the model:
```
type Batch struct {
        Spans   []*Span `protobuf:"bytes,1,rep,name=spans" json:"spans,omitempty"`
        Process Process `protobuf:"bytes,2,opt,name=process" json:"process"`
}
```
Considering the possibility of Zipkin spans, or even Istio spans with mixed services, the Process field should be optional: `*Process`. Should we make this change?  It's not an IDL change, just the Gogo annotations affecting Go struct.

Reply:
changed to 
```proto
message Batch {
    repeated Span spans = 1;
    Process process = 2 [
      (gogoproto.nullable) = true
    ];
}
```